packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -29,10 +34,33 @@ export const updateAdditionalWithdrawalData = async (
   return l2toL1TxWithDeadline
 }
 
+export async function attachTimestampToTokenWithdrawals({
+  withdrawals,
+  l2Provider
+}: {
+  withdrawals: WithdrawalInitiated[]
+  l2Provider: Provider
+}) {
+  return Promise.all(
+    withdrawals.map(async withdrawal => {
+      const txReceipt = await l2Provider.getTransactionReceipt(
+        withdrawal.txHash
+      )
+      const l2TxReceipt = new L2TransactionReceipt(txReceipt)
```

Comment:
For token withdrawal from event logs we are using `mapTokenWithdrawalFromEventLogsToL2ToL1EventResult` method which extracts the same transaction receipt + l2 to l1 event data for the transaction. Maybe we should merge this function into that so that we don't fetch tx receipt again and again, leading to more RPC calls than expected.

Or maybe this function should have an optional event prop which reuses the event from the previous extraction if already found.